Q&A Activity on Testing of Hypothesis

1.Carry out descriptive analytics on the dataset.
2.Check whether female and male students scored the same marks.
3.Whether test preparation helps the students?

In [57]:
import numpy as np
import pandas as pd
import scipy.stats as stats
from scipy.stats import ttest_ind

In [19]:
data=pd.read_csv('Week_8_Q&A_dataset - Sheet1.csv')
data.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

In [20]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Gender            28 non-null     object
 1   Test preparation  28 non-null     object
 2   Total Marks       28 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 800.0+ bytes


1.Carry out descriptive analytics on the dataset.

In [63]:
data.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

2. Check whether female and male students scored the same marks.

In [22]:
maledata=data[data['Gender']=='male']['Total Marks']
femaledata=data[data['Gender']=='female']['Total Marks']

In [23]:
maledata.describe()

count    12.000000
mean     30.416667
std      12.168950
min      14.000000
25%      21.750000
50%      29.000000
75%      43.250000
max      47.000000
Name: Total Marks, dtype: float64

In [24]:
femaledata.describe()

count    16.000000
mean     33.750000
std      12.865976
min      12.000000
25%      25.500000
50%      39.000000
75%      43.500000
max      50.000000
Name: Total Marks, dtype: float64

. The sample size is less than 30 ,we use t-test
. Taking Hypothesis: Ho:The average score of male and female are equal
                     Ha:The average score of male and female are not qual
 .Taking significance value as aplpha(α):0.05     

In [61]:

stat, p = ttest_ind(maledata,femaledata)
print('t=%.3f, p=%.3f' % (stat, p))


t=-0.694, p=0.494


There is a statistically significant difference in the average Total Marks between males and females, t= -0.694, p= 0.494.
The p value greater than alpha (0.494>0.05) 
Accepting the Null Hypothesis ,the average score of male and female are  equal

3. Whether test preparation helps the students?

In [43]:
completed=data[data['Test preparation']=='completed']['Total Marks']
notcompleted=data[data['Test preparation']=='none']['Total Marks']

In [44]:
completed.describe()

count     7.000000
mean     44.571429
std       3.359422
min      40.000000
25%      43.000000
50%      44.000000
75%      46.000000
max      50.000000
Name: Total Marks, dtype: float64

In [46]:
notcompleted.describe()

count    21.000000
mean     28.238095
std      11.661495
min      12.000000
25%      18.000000
50%      27.000000
75%      38.000000
max      48.000000
Name: Total Marks, dtype: float64

#Taking Hypothesis:
. Null Hypothesis Ho:Test preparation doesn't helps the students that is the average score of students who completed the testpreparation <= not completed the test preparation
. Alternative Hypothesis Ha:Test preparation helps the students (the average score of students who completed the testpreparation > not completed the test preparation)
. Taking significance value as aplpha(α):0.05
perform 1-tailed 2 sample test

In [62]:
stat, p = ttest_ind(completed,notcompleted)
print('t=%.3f, p=%.3f' % (stat, p))

t=3.614, p=0.001


Here, the p value is very small.The alpha value > p value so we can reject the Null Hypothesis.
Alternative Hypothesis is true that is, Test preparation helps the students.